server/main.js

```diff
@@ -0,0 +1,57 @@
+import Empirica from "meteor/empirica:core";
+
+import "./callbacks.js";
+import "./bots.js";
+
+import { taskData } from "./constants";
+
+// gameInit is where the structure of a game is defined.
+// Just before every game starts, once all the players needed are ready, this
+// function is called with the treatment and the list of players.
+// You must then add rounds and stages to the game, depending on the treatment
+// and the players. You can also get/set initial values on your game, players,
+// rounds and stages (with get/set methods), that will be able to use later in
+// the game.
+Empirica.gameInit(game => {
+  // Establish node list
+  console.log(game);
+
+  const nodes = [];
+  for (var i = game.players.length; i--; i > 0) nodes.push(i);
```

Comment:
-- i am not worried about the possibility of zero neighbors, BUT
-- i like the idea of sampling from other players
-- the reason i had defined 'nodes' was to set people on a path to a more sustainable way of doing it which (imho) is to pass in a graph defined elsewhere, rather than generate the graph in the javascript (among other reasons, so that it's not stochastic)
-- but this is definitely much cleaner for a demo,  and someone thinking about that would already know how to do it

Reply:
Totally agreed, networks should not be done like this eventually. Not sure we talked about this already, but we have tried to add some sort of simple network support into Empirica and realized we needed something better than what we had planned, we were really falling short. But we do eventually want to give some cool tools for game designers to generate and use networks. One day... 💃

Reply:
ah well i'd be happy to contribute to that effort!

a conversation for another page, but:  a very simple approach is to put a large TextArea input (as a config) that lets people put in an edgelist  (each row is a user, the first entry is their node, and the remaining entries are their neighbors, all separated by spaces or commas)

Reply:
For sure. That kind of thing can help create the graph. I can't recall all we wanted to do, but I remember we also wanted things like once the graph exists, you might want to edit it (say between rounds, depending on conditions), or you want to be able to traverse it (i.e. friends or friends), etc.

We talk about it again when we get more core stuff smoothed out. ⛵️

Reply:
this update is pushed